### PR TITLE
Add Polish to Missing Value Vizualizers in Contrib

### PR DESCRIPTION
**Pull request by ndanielsen**
This PR is addressing comments in the original PR for the missing values bar and dispersion visualizers. 

The original PR is located here: https://github.com/DistrictDataLabs/yellowbrick/pull/519

### CODE REVIEW

File: tests/test_contrib/test_missing/test_bar.py [hunk 1]
```diff
@@ -31,29 +31,37 @@
...
-    FeatureImportances visualizer
+    MissingBar Visualizer
     """
 
     def setUp(self):
-        super(TestMissingBarVisualizer, self).setUp()
+        super(MissingBarVisualizerTestCase, self).setUp()
         self.tol = 0.01
         if os.name == 'nt': # Windows
             self.tol = 0.5
```
Comment (ndanielsen) [inline tests/test_contrib/test_missing/test_bar.py]: In another PR, I'm going to move these tolerance settings into `VisualTestCase` so they can be more generally shared.

File: tests/test_contrib/test_missing/test_dispersion.py [hunk 2]
```diff
@@ -41,20 +41,25 @@ def setUp(self):
         if os.name == 'nt': # Windows
             self.tol = 5.0
 
+    @pytest.mark.skipif(pd is None, reason="test requires pandas")
```
Comment (bbengfort) [inline tests/test_contrib/test_missing/test_dispersion.py]: Same question as before - why does this test require pandas? Sorry if I'm missing something obvious!

File: tests/test_contrib/test_missing/test_dispersion.py [hunk 3]
```diff
@@ -107,18 +97,11 @@ def test_missingvaluesdispersion_with_numpy_with_y_targets(self):
...
-        X[X > 1.5] = np.nan
 
         features = [str(n) for n in range(20)]
         classes = ['Class A', 'Class B']
         viz = MissingValuesDispersion(features=features, classes=classes)
-        viz.fit(X, y=y)
+        viz.fit(self.missingdata.X, y=self.missingdata.y)
         viz.poof()
 
         self.assert_images_similar(viz, tol=self.tol)
```
Comment (bbengfort) [inline tests/test_contrib/test_missing/test_dispersion.py]: Awesome clean up of the tests - the fixture definitely made the tests much simpler to read!

File: yellowbrick/contrib/missing/bar.py [hunk 4]
```diff
@@ -95,31 +93,28 @@ def __init__(self, width=0.5, color='black', colors=None, classes=None, **kwargs
...
-            return nan_col_counts
+        if y is None:
+            self.nan_col_counts_ = [np.count_nonzero(np.isnan(col)) for col in nan_matrix.T]
+            return self.nan_col_counts_
 
         else:
             # add in counting of np.nan per target y by column
-            nan_counts = []
-            for target_value in np.unique(self.y):
+            for target_value in np.unique(y):
```
Comment (bbengfort) [inline yellowbrick/contrib/missing/bar.py]: Thank you for removing the data storage on the class! I think this is going to make the missing values visualizer much more performant!

**Comment by ndanielsen** [comment]
@bbengfort my personal criteria for hoisting these visualizers into the main package are:
- For bar viz, adding an option for percent in addition to count when targets are supplied
- For dispersion, making sure that it accepts datetime index
- Adding the quick method into the documentation. I think that we likely should emphasize quick methods more especially for feature vizualizers

**Comment by ndanielsen** [comment]
I'm going to address the comments here and then I'll merge in. @bbengfort thanks as always for the great code review!

**Comment by bbengfort** [comment]
@ndanielsen those sound like good criteria and I'm excited that we're close to getting them into the main package! 

What would you think about putting these visualizers under a new package called `yellowbrick.wrangling`? Potentially this falls under feature analysis, but I think this along with outlier detection visualizers might be something that would fall under its own package? 

As for the quick methods, I agree that these should be hoisted to prime time. See #600 which is already underway in #669. These issues change the quick method API a tad (e.g. returning a visualizer) but also ensure consistency across all quick methods. Once that is done, we need to find a way to include them more informatively in the documentation. I'm not sure whether to autodoc them or to create a special section for them (or a special format in the API documentation for them). 

Also one quick comment; see #673 -- there might be a minor merge collision with this PR. Would you mind working with @Kautumn06 to make sure both PRs are merged easily?

**Comment by rebeccabilbro** [comment]
Hey there @ndanielsen - is this something you're still interested in working on? It's just occurring to me that the develop branch is starting to diverge somewhat significantly from your PR branch (and it's about to change a lot more with #687, #669, etc), so wanted to double check with you before things get too tangly!

**Comment by rebeccabilbro** [comment]
Hi @ndanielsen! Given that the `develop` branch has evolved enough that we'll probably end up with some fairly gnarly conflicts if we try to merge this in now, I'm going to go ahead and archive this PR. Thank you for all the work you put into this and hope we can revisit these contributions sometime in the future; I know it will be a very popular feature!

**Pull request closed without merging**